impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -50,6 +50,7 @@
   // rendering but also creates smaller intermediate passes.
   ISize root_size;
   if (requires_blit) {
+    ASSERT_OPTIONAL_HAS_VALUE(clip_rect);
```

Comment:
This is fine if there's unit test coverage that will cause the assert to trigger if someone violates the assumptions that make the assert correct today. Otherwise we're no better off than we were when we didn't have the clang-tidy lint running.

If the test that covers the assumptions is at a different level, then should this instead be this?
```C++
if (!clipt_rect.has_value()) {
  VALIDATION_LOG << "...";
  return std::nullopt;
}
```

That way this code can be defensive against someone calling `WrapTextureWithRenderTarget` incorrectly by accident.

Reply:
If `requires_blit` is set, then `clip_rect` must always be set and there is no fallback path.

The assert macro using `FML_UNREACHABLE` is intended to represent that the `!has_value()` case should never happen here.

Reply:
I'm somewhat biased towards returning an empty result in these particular cases.

On the one hand, we've heavily biased towards returning null resources/error flags for clear "this should never happen" cases across render routines, which normally results in a frame getting skipped. That's actually saved us from needlessly crashing in some cases where a user's app might just get a janky animation due to some skipped frames that are triggering an unexpected case.
On the other hand, it's often quite annoying to find these sources of error, and an explicit crash would be quicker to pinpoint. However, that issue is mostly solved through validation logging, which is largely missing here in `SurfaceMTL` at the moment.